### PR TITLE
Change associative array to use ID instead of name.  

### PR DESCRIPTION
**Pull request by stillema**
Allow -E option to match on either name or UUID for environment.

### CODE REVIEW

File: config/settings.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func (s FileSettingsRetriever) GetSettings(envName, svcName, accountsHost, authH
 func StoreEnvironments(envs *[]models.Environment, settings *models.Settings) {
 	settings.Environments = map[string]models.AssociatedEnvV2{}
 	for _, env := range *envs {
-		settings.Environments[env.Name] = models.AssociatedEnvV2{
+		settings.Environments[env.ID] = models.AssociatedEnvV2{
```
Comment (forana) [inline config/settings.go]: won't this break for anyone with an already-existing settings file?

Comment (stillema) [reply config/settings.go]: I did some searching around for how this associative key is used and it doesn't appear to affect anything else.

However I noticed I forgot to update the migrate settings method to create the associative array with the ID as key.  I updated that now too.  

Also updated the `SetGivenEnv` to better match the variable meaning.

Comment (forana) [reply config/settings.go]: nvm, I see what's happening.

**Comment by forana** [comment]
LGTM

**Pull request merged**